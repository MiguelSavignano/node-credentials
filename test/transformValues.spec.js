const { transformValues } = require('../src/transformValues');
require('./helpers/matchers');
const complexJSON = require('./examples/transformValues/complex.json');

describe('core', () => {
  test('transformValues sync', () => {
    const result = transformValues(complexJSON, (value) => `${value} MOCK`);
    expect(result).toMatchSnapshot();
  });

  test('transformValues async', async () => {
    const result = await transformValues(complexJSON, async (value) => `${value} MOCK`);
    expect(result).toMatchSnapshot();
  });
});
