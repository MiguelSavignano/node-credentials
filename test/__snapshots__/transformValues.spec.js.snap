// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core transformValues async 1`] = `
Object {
  "apikey": "1234 MOCK",
  "complex": Array [
    Array [
      "1 MOCK",
      Object {
        "data": Array [
          Object {
            "a": "4 MOCK",
            "b": Array [
              Array [
                "8 MOCK",
              ],
            ],
          },
        ],
      },
    ],
  ],
  "config": Array [
    Object {
      "data": "data MOCK",
    },
    Object {
      "data": "data2 MOCK",
    },
    "12345 MOCK",
    "78 MOCK",
  ],
  "data2": null,
  "database": Object {
    "password": "12345 MOCK",
    "user": "1234 MOCK",
  },
  "error": null,
  "errors": Array [
    Array [
      "1 MOCK",
      "2 MOCK",
      "3 MOCK",
    ],
  ],
  "keys": Array [
    "1234 MOCK",
    "123 MOCK",
  ],
  "password": "12344 MOCK",
}
`;

exports[`core transformValues sync 1`] = `
Object {
  "apikey": "1234 MOCK",
  "complex": Array [
    Array [
      "1 MOCK",
      Object {
        "data": Array [
          Object {
            "a": "4 MOCK",
            "b": Array [
              Array [
                "8 MOCK",
              ],
            ],
          },
        ],
      },
    ],
  ],
  "config": Array [
    Object {
      "data": "data MOCK",
    },
    Object {
      "data": "data2 MOCK",
    },
    "12345 MOCK",
    "78 MOCK",
  ],
  "data2": null,
  "database": Object {
    "password": "12345 MOCK",
    "user": "1234 MOCK",
  },
  "error": null,
  "errors": Array [
    Array [
      "1 MOCK",
      "2 MOCK",
      "3 MOCK",
    ],
  ],
  "keys": Array [
    "1234 MOCK",
    "123 MOCK",
  ],
  "password": "12344 MOCK",
}
`;
